clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/ClusterProvider.groovy

```diff
@@ -95,4 +95,11 @@ interface ClusterProvider<T extends Cluster> {
    * @return the identifier of the backing cloud provider
    */
   String getCloudProviderId()
+
+  /**
+   * Determines whether or not optimizations can be made by retrieving minimal or unexpanded clusters.
+   *
+   * This primarily affects how server groups are loaded for a cluster (@see com.netflix.spinnaker.clouddriver.controllers.ClusterController}.
+   */
+  boolean supportsMinimalClusters()
```

Comment:
If this were a Java interface with a default method for this, would Groovy accept that and prevent you from having to touch all the cloud providers?

Not sure that's better - might be best to make it explicit.

Reply:
That is 100% ... in the interest of not messing around I didn't convert the interface.

Reply:
Groovy has traits that are similar to java 8 default interface methods. They were handy in the cases I've used them.